opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ServiceContextProvider.java

```diff
@@ -64,15 +74,41 @@
         if (null != authSession) {
             return authSession.getParent();
         } else {
-            ServiceSession session = new ServiceSession();
+            FacadeServiceableRequest facadeServiceable = request.getFacadeServiceable();
+            String sessionPassword = facadeServiceable.getSessionPassword();
+
+            if (facadeServiceable.getXRequestID() == null) {
+                throw new RuntimeException("no request id");
+            }
 
-            if (null != request.getFacadeServiceable().getServiceSessionId()) {
-                session.setId(request.getFacadeServiceable().getServiceSessionId());
+            Optional<ServiceSession> savedSession = serviceSessions.findById(facadeServiceable.getXRequestID());
+            if (savedSession.isPresent()) {
+                ServiceSession serviceSession = savedSession.get();
+                if (StringUtils.isBlank(new String(serviceSession.getPassword()))) {
+                    if (StringUtils.isBlank(sessionPassword)) {
+                        throw new RuntimeException("No password. Password has to be provided in request or saved in db");
+                    }
+                    serviceSession.setPassword(sessionPassword.getBytes());
+                }
+                return serviceSession;
             }
 
-            session.setContext(MAPPER.writeValueAsString(request));
+            if (sessionPassword == null) {
```

Comment:
It is required only on the first request. I don't see where you update it from a database for subsequent calls

Reply:
Now I store secretkey in db but when secretkey is cleared then password required again